core/snippets/snippets.py

```diff
@@ -0,0 +1,149 @@
+import glob
+from pathlib import Path
+
+from talon import Context, Module, app, fs
+
+from ..modes.language_modes import language_ids
+from .snippet_types import Snippet
+from .snippets_parser import create_snippets_from_file
+
+SNIPPETS_DIR = Path(__file__).parent / "snippets"
+
+mod = Module()
+
+mod.list("snippet_insert", "List of insertion snippets")
+mod.list(
+    "snippet_insert_with_phrase", "List of insertion snippets containing a text phrase"
+)
+mod.list("snippet_wrap", "List of wrapper snippets")
+
+context_map = {
+    "_": Context(),
+}
+snippets_map = {}
+
+# Create a context for each defined language
+for lang in language_ids:
+    ctx = Context()
+    ctx.matches = f"code.language: {lang}"
+    context_map[lang] = ctx
+
+
+@mod.action_class
+class Actions:
+    def get_snippet(name: str) -> Snippet:
+        """Get snippet by <name>"""
+        return snippets_map[name]
+
+
+def update_snippets():
+    grouped = group_by_language(get_snippets())
+
+    snippets_map.clear()
+
+    for lang, ctx in context_map.items():
+        insertion_map = {}
+        insertions_phrase_map = {}
+        wrapper_map = {}
+
+        for fl in get_fallback_languages(lang):
+            snippets, insertions, insertions_phrase, wrappers = create_lists(
+                lang, fl, grouped.get(fl, [])
+            )
+            snippets_map.update(snippets)
+            insertion_map.update(insertions)
+            insertions_phrase_map.update(insertions_phrase)
+            wrapper_map.update(wrappers)
+
+        ctx.lists["user.snippet_insert"] = insertion_map
+        ctx.lists["user.snippet_insert_with_phrase"] = insertions_phrase_map
+        ctx.lists["user.snippet_wrap"] = wrapper_map
+
+
+def get_snippets() -> list[Snippet]:
+    files = glob.glob(f"{SNIPPETS_DIR}/*.snippet")
+    result = []
+
+    for file in files:
+        result.extend(create_snippets_from_file(file))
+
+    return result
+
+
+def get_fallback_languages(language: str) -> list[str]:
+    match language:
```

Comment:
At some point we'll prob want to make it easier for users to configure the hierarchy here, but probably ok to just have it baked in and let them tweak this function / file PRs for now

Reply:
Agreed

Reply:
Up to you whether to add this to follow-up issue or just leave it and cross that bridge if we need to later